go-controller/pkg/node/healthcheck.go

```diff
@@ -93,29 +96,120 @@ func countLocalEndpoints(ep *kapi.Endpoints, nodeName string) int {
 	return num
 }
 
-// check for OVS internal ports without any ofport assigned, they are stale ports that must be deleted
-func checkForStaleOVSInterfaces(stopChan chan struct{}) {
-	for {
-		select {
-		case <-time.After(60 * time.Second):
-			stdout, _, err := util.RunOVSVsctl("--data=bare", "--no-headings", "--columns=name", "find",
-				"interface", "ofport=-1")
+// checkForStaleOVSInternalPorts checks for OVS internal ports without any ofport assigned,
+// they are stale ports that must be deleted
+func checkForStaleOVSInternalPorts() {
+	stdout, _, err := util.RunOVSVsctl("--data=bare", "--no-headings", "--columns=name", "find",
+		"interface", "ofport=-1")
+	if err != nil {
+		klog.Errorf("Failed to list OVS interfaces with ofport set to -1")
+		return
+	}
+	if len(stdout) == 0 {
+		return
+	}
+
+	values := strings.Split(stdout, "\n\n")
+	for _, val := range values {
+		klog.Warningf("Found stale interface %s, so deleting it", val)
+		_, stderr, err := util.RunOVSVsctl("--if-exists", "--with-iface", "del-port", val)
+		if err != nil {
+			klog.Errorf("Failed to delete OVS port/interface %s: stderr: %s (%v)",
+				val, stderr, err)
+		}
+	}
+}
+
+// checkForStaleOVSRepresentorInterfaces checks for stale OVS ports backed by Repreresentor interfaces,
+// derrive iface-id from pod name and namespace then remove any interfaces assoicated with a sandbox that are
+// not scheduled to the node.
+func checkForStaleOVSRepresentorInterfaces(nodeName string, k kube.Interface) {
+	// list Ports on br-int and find out the current ports that have iface-ids
+	currentIfaceIds := make(map[string]string)
+	ports, stderr, err := util.RunOVSVsctl("list-ports", "br-int")
+	if err != nil {
+		klog.Errorf("failed to get list of ports on br-int:, stderr: %q, error: %v", stderr, err)
+		return
+	}
+
+	for _, port := range strings.Split(ports, "\n") {
+		stdout, stderr, err := util.RunOVSVsctl("get", "Port", port, "Interfaces")
+		if err != nil {
+			klog.Errorf("failed to get port %q on br-int:, stderr: %q, error: %v", port, stderr, err)
+			continue
+		}
+		// remove brackets on list of interfaces
+		ifaces := strings.TrimPrefix(strings.TrimSuffix(stdout, "]"), "[")
```

Comment:
I'd rather not have all this string parsing. Can we just use ovs-vsctl -f json get Port port Interfaces and unmarshal into correct data structures?

Reply:
sounds solid.

This was just a pattern i saw in `nicstobridge.go` 
working with json is better.